Add /v4 to references to dafny-lang/DafnyRuntimeGo

robin-aws opened the pull request:
### Description
We fixed the github.com/dafny-lang/DafnyRuntimeGo repository by adding a copy under `v4/`, but need to update all of the references to that path in Dafny itself.

### How has this been tested?
Existing tests of Go using `--go-module-name`. Also added an explicit test against the published runtime, since the existing ones use `replace` directives to test against the local copy in `DafnyRuntimeGo-gomod` instead.

I also found for this test to work, I had to push a new tag on `github.com/dafny-lang/DafnyRuntimeGo`, since the checksum didn't match on the existing `v4.7.0` tag. I suspect we moved the tag at some point when debugging the issues, which does not work since the Go checksum repository can't be updated. I went with `v4.8.0` just in case it still didn't work and we needed to make extra changes and then tag with `v4.8.1`. 

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

ShubhamChaturvedi7 commented:
We should have a separate integration tests which run against the published version. `DafnyRuntimeGo-gomod` was added because we didn't have a published library back then. We should run local tests against a copy which is generated by Dafny, and afaik `DafnyRuntimeGo-gomod` is static.

robin-aws commented:
> We should have a separate integration tests which run against the published version. `DafnyRuntimeGo-gomod` was added because we didn't have a published library back then. We should run local tests against a copy which is generated by Dafny, and afaik `DafnyRuntimeGo-gomod` is static.

I realized this was worth doing and not hard, the only caveat was potentially needing to publish a new version of the runtime before a new version of Dafny itself, but I think that's worth it. Added a test and updated the PR description.
